### PR TITLE
User defined modules salt

### PR DESCRIPTION
**Pull request by Buckram123**
This PR aims to add control of instantiate2 salt for the modules
### Checklist

- [x] CI is green.
- [x] Changelog updated.

### CODE REVIEW

File: framework/packages/abstract-core/src/native/account_factory.rs [hunk 1]
```diff
@@ -104,6 +104,8 @@ pub enum ExecuteMsg {
         namespace: Option<String>,
         // Provide list of module to install after account creation
         install_modules: Vec<ModuleInstallConfig>,
+        // Provide module salt, that will be added to instantiate2 salt of installed modules
+        module_salt: Option<Binary>,
```
Comment (CyberHoward) [inline framework/packages/abstract-core/src/native/account_factory.rs:108]: If also used as salt for account it should be renamed to `instantiate_salt` or similar. Preferably also renamed in the manager API.

File: framework/packages/abstract-interface/src/account/manager.rs [hunk 2]
```diff
@@ -65,22 +65,31 @@ impl<Chain: CwEnv> Manager<Chain> {
     pub fn replace_api(
         &self,
         module_id: &str,
+        // this field is required if you reinstall same version
+        module_salt: Option<Binary>,
```
Comment (CyberHoward) [inline framework/packages/abstract-interface/src/account/manager.rs:69]: replace_api is actually replace_adapter and adapters never need salts (as they are already instantiated)

File: framework/packages/abstract-interface/src/native/account_factory.rs [hunk 3]
```diff
@@ -62,6 +63,7 @@ impl<Chain: CwEnv> AccountFactory<Chain> {
             namespace,
             base_asset,
             install_modules,
+            module_salt,
```
Comment (CyberHoward) [inline framework/packages/abstract-interface/src/native/account_factory.rs:66]: Let's just replace all "module_salt" with "instantiate_salt"

**Comment by Buckram123** [comment]
Closed in favor of #243

**Pull request closed without merging**